core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -546,3 +546,78 @@ func unbroadcastAttempt(db *gorm.DB, a models.EthTxAttempt) error {
 	}
 	return errors.Wrap(db.Exec(`UPDATE eth_tx_attempts SET broadcast_before_block_num = NULL, state = 'in_progress' WHERE id = ?`, a.ID).Error, "unbroadcastAttempt failed")
 }
+
+// ForceRebroadcast sends a transaction for every nonce in the given nonce range at the given gas price.
+// If an eth_tx exists for this nonce, we re-send the existing eth_tx with the supplied parameters.
+// If an eth_tx doesn't exist for this nonce, we send a zero transaction.
+// This operates completely orthogonal to the normal EthConfirmer and can result in untracked attempts!
+// Only for emergency usage.
+// Deliberately does not take the advisory lock (we don't write to the database so this is safe from a data integrity perspective).
+// This is in case of some unforeseen scenario where the node is refusing to release the lock. KISS.
+func (ec *ethConfirmer) ForceRebroadcast(beginningNonce uint, endingNonce uint, gasPriceWei uint64, address gethCommon.Address, overrideGasLimit uint64) error {
+	logger.Info("ForceRebroadcast: will rebroadcast transactions for all nonces between %v and %v", beginningNonce, endingNonce)
+
+	for n := beginningNonce; n <= endingNonce; n++ {
+		etx, err := findEthTxWithNonce(ec.store.GetRawDB(), address, n)
+		if err != nil {
+			return errors.Wrap(err, "ForceRebroadcast failed")
+		}
+		if etx == nil {
+			logger.Debugf("ForceRebroadcast: no eth_tx found with nonce %v, will rebroadcast empty transaction", n)
+			hash, err := ec.sendEmptyTransaction(address, n, overrideGasLimit, gasPriceWei)
```

Comment:
Just so I understand why you sending an empty transaction here:
Is the idea here to "consume" the nonce so that later transactions with a higher nonce get unblocked? 
Is it not the case that even an "empty" tx still costs gas to add to a block? 
How do we know there are other transactions that are worth the cost? Afaict, it's the user that sets the nonce range. 
How do we get into situations where a nonce is not used for an address? When the gas value is too small!? But the node would tell us, right?

Reply:
>  Is the idea here to "consume" the nonce so that later transactions with a higher nonce get unblocked?

Yes.

>  Is it not the case that even an "empty" tx still costs gas to add to a block?

Yes, it costs gas.

>  How do we know there are other transactions that are worth the cost? Afaict, it's the user that sets the nonce range.

Don't understand the question.

>  How do we get into situations where a nonce is not used for an address? When the gas value is too small!? But the node would tell us, right?

It should never happen. I cannot imagine a scenario when it would happen. However, things that I cannot imagine have happened before, so this command is included as an emergency fallback in case we need to clear a stuck transaction queue fast, however it got there.